avm/res/network/network-manager/main.bicep

```diff
@@ -240,3 +238,158 @@ type roleAssignmentType = {
   delegatedManagedIdentityResourceId: string?
 }[]?
+
+type networkGroupType = {
+  @sys.description('Required. The name of the network group.')
+  name: string
+
+  @sys.description('Optional. A description of the network group.')
+  description: string?
+
+  @sys.description('Optional. Static Members to create for the network group. Contains virtual networks to add to the network group.')
+  staticMembers: {
+    @sys.description('Required. The name of the static member.')
+    name: string
+
+    @sys.description('Required. Resource ID of the virtual network.')
+    resourceId: string
+  }[]?
+}[]?
+
+type networkManagerScopeAccessType = ('Connectivity' | 'SecurityAdmin')[]
+
+type networkManagerScopesType = {
+  @sys.description('Optional. Management groups to assign to the network manager to manage.')
+  managementGroups: string[]?
+
+  @sys.description('Optional. Subscriptions to assign to the network manager to manage.')
+  subscriptions: string[]?
+}
+
+type scopeConnectionType = {
+  @sys.description('Required. The name of the scope connection.')
+  name: string
+
+  @sys.description('Optional. A description of the scope connection.')
+  description: string?
+
+  @sys.description('Required. Enter the subscription or management group resource ID that you want to add to this network manager\'s scope.')
+  resourceId: string
+
+  @sys.description('Required. Tenant ID of the subscription or management group that you want to manage.')
+  tenantId: string
+}[]?
+
+type connectivityConfigurationsType = {
+  @sys.description('Required. The name of the connectivity configuration.')
+  name: string
+
+  @sys.description('Optional. A description of the connectivity configuration.')
+  description: string?
+
+  @sys.description('Required. Network Groups for the configuration. A connectivity configuration must be associated to at least one network group.')
+  appliesToGroups: {
+    @sys.description('Required. Group connectivity type.')
+    groupConnectivity: ('DirectlyConnected' | 'None')
+
+    @sys.description('Optional. Flag if global is supported.')
+    isGlobal: ('True' | 'False')?
+
+    @sys.description('Required. Network group Id.')
+    networkGroupId: string
+
+    @sys.description('Optional. Flag if use hub gateway.')
+    useHubGateway: ('True' | 'False')?
+  }[]
+
+  @sys.description('Required. The connectivity topology to apply the configuration to.')
+  connectivityTopology: ('HubAndSpoke' | 'Mesh')
+
+  @sys.description('Optional. The hubs to apply the configuration to.')
+  hubs: {
+    @sys.description('Required. Resource Id of the hub.')
+    resourceId: string
+
+    @sys.description('Required. Resource type of the hub.')
+    resourceType: 'Microsoft.Network/virtualNetworks'
+  }[]?
+
+  @sys.description('Optional. Delete existing peering connections.')
+  deleteExistingPeering: ('True' | 'False')?
+
+  @sys.description('Optional. Is global configuration.')
+  isGlobal: ('True' | 'False')?
```

Comment:
There is actually a new [spec](https://github.com/Azure/Azure-Verified-Modules/pull/803) waiting in a PR that asks primitive types to be declared as such (e.g. use int instead if string for numbers, or boolean instead of string for true/false).
This would mean, this parameter should be changed to `isGlobal: bool` and the place where it is used needs to (optionally) cast the value to a string. I've seen implicit casts in the pasts, so we may not need it.

Same comment for the other string-boolean parameters

Reply:
Interesting. I would avoid hiding such things and align to the API as much as possible. That way its a standard consumption method from an API and IaC perspective and less abstraction code to maintain. But if that's the path that AVM is going to take then I will wait for that PR to be merged. Would be great to know if that's something that will be as a recommendation for the module owner or something that is a must have.

Reply:
I'd argue it's probably a 'Should' that targets improved usability of the resource, which in many instances is just plain bad (case an point CMK or boolean strings 😄 )

Reply:
Hey @AlexanderSehr I have done some testing on the casting approach.

![image](https://github.com/Azure/bicep-registry-modules/assets/28486158/5a3406c2-2951-4b3a-ac8f-0292fb60107c)

Here is an [example PR](https://github.com/ahmadabdalla/bicep-registry-modules/pull/14/files#diff-2b0d9ee4256af0d2862695cca181bbca439dd6beb31013e1bb33c8b0b8b9f546R44): comparing the current UDT implementation as per this PR with an approach for casting.

I def took a toll on testing and validation a lot more than it was before. And seems more code to maintain. What are your thoughts on this?

I have some questions around the `default` behaviour for a UDT param of type `bool`. Because I had to use casting, I now am enforced in my test files to explicitly pass the parameter either `true` `false` compared to the past.  Is there a specific pattern I should follow here as per AVM standards?

![image](https://github.com/Azure/bicep-registry-modules/assets/28486158/10166549-bb15-4771-9830-2503755743fa)

See above picture as an example. If I do not provide these values, my deployments now fail :/

Reply:
Hey @ahmadabdalla,
maybe we can find a slot some time to actually have a chat about the subject, as it's just easier to talk about than write 😄 
Anyhow, the modules I've seen implement this on scale did usually not introduce much added code. Usually it was not much more than introducing e.g. a `string(parameter)` or `'${parameter}'` where a parameter was used an a string was expected, while in others it made the module actually easier as conditions didn't need to check for explicit string values in conditions like `if(parameter == 'true')`. But naturally, the argument you made (i.e., deviating from the API) was exactlythe discussion to have. In like 'is it worth it'. At least for now we opted for 'yes' on the backdrop that we want to improve/easy a deployments usuability when we can.

Regarding the default behavior - if a parameter is optional, you can still just make it nullable. Also booleans. Does this answer the question?

Reply:
Got everything working now. Ready for review